### PR TITLE
Implement ToString for Variant

### PR DESCRIPTION
**Pull request by annelo-msft**
Addresses https://github.com/Azure/azure-sdk-for-net/issues/32978

### CODE REVIEW

File: sdk/core/Azure.Core.Experimental/src/Variant/Variant.cs [hunk 1]
```diff
@@ -1388,5 +1388,41 @@ private Variant(object o, ulong u)
...
+            }
+            else if (_object is TypeFlag flag)
+            {
+                if (Type == typeof(bool))
+                {
+                    result = flag.ToObject(this).ToString()!.ToLowerInvariant();
+                }
+                else
+                {
+                    result = flag.ToObject(this).ToString();
```
Comment (KrzysztofCwalina) [inline sdk/core/Azure.Core.Experimental/src/Variant/Variant.cs:1409]: this should also be optimized for common primitives. Feel free to improve it in a subsequent release.

Comment (annelo-msft) [reply sdk/core/Azure.Core.Experimental/src/Variant/Variant.cs:1409]: Are you thinking we would add a switch statement on Type and use that to not box the known value-type primitives?

File: sdk/core/Azure.Core.Experimental/tests/Variant/VariantUsage.cs [hunk 2]
```diff
@@ -28,5 +32,61 @@ public void CanTestForNull()
 
             Assert.True(new Variant((object)null).IsNull);
         }
+
+        //[Test]
```
Comment (christothes) [inline sdk/core/Azure.Core.Experimental/tests/Variant/VariantUsage.cs:36]: intentional?

Comment (annelo-msft) [reply sdk/core/Azure.Core.Experimental/tests/Variant/VariantUsage.cs:36]: No, thank you for catching!

File: sdk/core/Azure.Core.Experimental/src/Variant/Variant.cs (hunk 1 above)
Comment (annelo-msft) [reply sdk/core/Azure.Core.Experimental/src/Variant/Variant.cs]: Done.

**Review by KrzysztofCwalina — APPROVED** [review]

**Pull request merged**